src/main/java/novamachina/exnihilosequentia/common/block/BlockSieve.java

```diff
@@ -115,7 +116,12 @@ public BlockState getStateForPlacement(BlockPlaceContext context) {
     public void playerDestroy(@Nonnull Level worldIn, @Nonnull Player player, @Nonnull BlockPos pos, @Nonnull BlockState state, @Nullable BlockEntity te, @Nonnull ItemStack stack) {
         super.playerDestroy(worldIn, player, pos, state, te, stack);
         if (!worldIn.isClientSide() && te instanceof SieveTile) {
-            ((SieveTile) te).removeMesh(false);
+            SieveTile sieveTile = (SieveTile) te;
+            sieveTile.removeMesh(false);
+            if (!sieveTile.getBlockStack().isEmpty()) {
+                worldIn.addFreshEntity(new ItemEntity(worldIn, pos.getX() + 0.5F, pos.getY() + 1.1F, pos
+                        .getZ() + 0.5F, sieveTile.getBlockStack()));
+            }
```

Comment:
SieveTile#removeMesh already adds the mesh to the world.

Reply:
this doesn't drop mesh, it's dropping the block inside when sieving :)